File: Exam/bai1.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main()
{
    vector<int> num_list{3, 4, 5, 1, -44, 5, 10, 12, 33, 1};
    int k = 3;
    vector<int> result;

    for (int i = 0; i <= num_list.size() - k; i++)
    {
        vector<int> sub_list(num_list.begin() + i, num_list.begin() + i + k);
        result.push_back(*max_element(sub_list.begin(), sub_list.end()));
    }

    for (int i = 0; i < result.size(); i++)
    {
        cout << result[i] << " ";
    }
    cout << endl;

    return 0;
}

File: Exam/bai1.py
num_list = [3, 4, 5, 1, -44 , 5 ,10, 12 ,33, 1]
k=3
start_index = list(range(0, len(num_list)-k+1))
end_index = list(range(k, len(num_list)+1))
result = []
for start, end in zip(start_index, end_index):
    sub_list = num_list[start:end]
    result.append(max(sub_list))
print(result)
